test/jdk/javax/net/ssl/DTLS/InvalidRecords.java

```diff
@@ -36,27 +36,32 @@
 
 import java.net.DatagramPacket;
 import java.net.SocketAddress;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
  * Test that if handshake messages are crasged, the handshake would fail
  * because of handshaking hash verification.
  */
 public class InvalidRecords extends DTLSOverDatagram {
-    boolean needInvalidRecords = true;
+    private static final AtomicBoolean needInvalidRecords = new AtomicBoolean(true);
 
     public static void main(String[] args) throws Exception {
         InvalidRecords testCase = new InvalidRecords();
         testCase.runTest(testCase);
-    }
 
-    @Override
-    public boolean isGoodJob() {
-        return false;
+        if (needInvalidRecords.get()) {
+            // if this is true, the createHandshakePacket() method
+            // was NOT called twice to create ClientHello messages
+            throw new RuntimeException(
+                    "The invalid handshake packet was not"
+                    + " rejected as it should have been.");
+        }
     }
 
+
     @Override
     DatagramPacket createHandshakePacket(byte[] ba, SocketAddress socketAddr) {
-        if ((ba.length >= 60) &&
+        if (needInvalidRecords.get() && (ba.length >= 60) &&
                 (ba[0x00] == (byte)0x16) && (ba[0x0D] == (byte)0x01) &&
                 (ba[0x3B] == (byte)0x00) && (ba[0x3C] > 0)) {
```

Comment:
I just want to make sure - this test is only designed to be run for initial handshakes with cookies, not resumed handshakes, correct?  I assume that is the intent since this test dates back to the initial DTLS release where resumptions didn't use cookies (that was a recent change to include support for resumption cookies).

Reply:
That's correct. The test driver (DTLSOverDataGram) doesn't do session resumption.